clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -3099,3 +3109,25 @@ mlir::cir::SourceLanguage CIRGenModule::getCIRSourceLanguage() {
   // TODO(cir): support remaining source languages.
   llvm_unreachable("CIR does not yet support the given source language");
 }
+
+LangAS CIRGenModule::GetGlobalVarAddressSpace(const VarDecl *D) {
+  if (langOpts.OpenCL) {
+    LangAS AS = D ? D->getType().getAddressSpace() : LangAS::opencl_global;
```

Comment:
Note that `VarDecl *D` can also be specified as `nullptr`. This case often occurs for globals generated by compiler itself, e.g. C++ ABI. Irrelevant to our changes.